diff --git a/lib/wind/generate.py b/lib/wind/generate.py
index 1d708c6..8df3a81 100644
--- a/lib/wind/generate.py
+++ b/lib/wind/generate.py
@@ -35,7 +35,6 @@
 # SUCH DAMAGE. 
 
 import datetime
-import string
 import os
 
 class GeneratedFile :
@@ -56,10 +55,10 @@ class GeneratedFile :
 
 class Header(GeneratedFile) :
     "Represents a generated header file"
-    guardTrans = string.maketrans('-.', '__')
+    guardTrans = str.maketrans('-.', '__')
     def makeGuard(self) :
         """Return a name to be used as ifdef guard"""
-        return string.upper(string.translate(self.name, self.guardTrans))
+        return str.upper(str.translate(self.name, self.guardTrans))
 
     def __init__(self, name) :
         "Create a new Header with name"
diff --git a/lib/wind/stringprep.py b/lib/wind/stringprep.py
index 249b1dc..1d038f2 100644
--- a/lib/wind/stringprep.py
+++ b/lib/wind/stringprep.py
@@ -57,7 +57,7 @@ def symbols(tabledict, tables):
         list = list + tabledict.get(x, [])
     if len(list) == 0:
         return ""
-    return "|".join(map(lambda x: "WIND_PROFILE_%s" % (string.upper(x)), list))
+    return "|".join(map(lambda x: "WIND_PROFILE_%s" % (str.upper(x)), list))
 
 def get_errorlist():
     d = dict()
